ImperatorToCk2/src/main/java/ImperatorToCK2/Processing.java

```diff
@@ -1,707 +1,716 @@
+    {
+
+        String[] toAdd = convertProvConvList("provinceConversion2.txt","provinceConversion2Formatted.txt");
+        String VM = "\\"; 
+        VM = VM.substring(0);
+
+        FileInputStream fileIn= new FileInputStream(source);
+        Scanner scnr= new Scanner(fileIn);
+
+        FileOutputStream fileOut= new FileOutputStream(destination);
+        PrintWriter out = new PrintWriter(fileOut);
+
+        String qaaa = scnr.nextLine();
+
+        int aqq = 0;
+
+        while (aqq < toAdd.length) {
+            if (toAdd[aqq] != null) {
+                out.println(toAdd[aqq]); 
+            }
+            aqq = aqq + 1; 
+        }
+
+        int flag = 0;
+        try {
+            while (flag == 0) {
+                out.println(qaaa);
+                qaaa = scnr.nextLine();
+
+            }
+
+        }catch (java.util.NoSuchElementException exception){
+            flag = 1; 
+            out.flush();
+            fileOut.close();
+        }
+
+        return 0;
+    }
+
+    public static ArrayList<String> generateSubjectList(int tot, String source) throws IOException
+    {
+
+        ArrayList<String> subjects = new ArrayList<String>();
+
+        subjects.add("0,0,feudatory"); //Debug at id 0 so list will never be empty
+
+        int aqq = 0;
+
+        while (aqq < tot) {
+
+            //String[] relation = Importer.importSubjects(source,aqq,subjects);
+            subjects = Importer.importSubjects(source,aqq,subjects);
+            //if (relation[0] != "9999") {
+            //subjects.add(relation[0]+","+relation[1]+","+relation[2]);
+
+            //}
+            aqq = aqq + 1; 
+        }
+
+        return subjects;
+    }
+
+    public static int checkSubjectList(int country, ArrayList<String> subjects) throws IOException
+    {
+
+        String countryStr = Integer.toString(country);
+        int aqq = 0;
+
+        int output = 9999;//default, if 9999, country remains free
+
+        while (aqq < subjects.size()) {
+
+            String subCountry = subjects.get(aqq).split(",")[1];
+
+            if (countryStr.equals(subCountry)) {
+                output = aqq;
+                aqq = subjects.size() + 1000; //end loop
+            }
+            aqq = aqq + 1; 
+        }
+
+        return output;
+    }
+
+    public static int checkMonumentList(String name) throws IOException //Checks if save is from 2.0+ or before
+    {
+
+        FileInputStream fileIn= new FileInputStream(name);
+        Scanner scnr= new Scanner(fileIn);
+        String qaaa = scnr.nextLine();
+
+        int flag = 0;
+
+        if (qaaa.equals ("great_work_manager={")) {
+            flag = 1;
+
+        }
+
+        return flag;
+    }
+
+    public static String formatSaveName(String name) {// replaces /:*?"<>|- and space with _, while removing tabs and .rome
+        String quote = '"'+""; // " character, Java doesn't like isoated " characters
+        String tab = "    ";
```

Comment:
This is actually a string of 4 spaces. I doubt you intended this?

Reply:
Indeed, lol, my IDE replaced hard tab characters with spaces. I'll try to upload a corrected version sometime tonight or tomorrow